File: Ontology/ObjectPropertyDomain.cs
namespace Ontology
{
    public class ObjectPropertyDomain:
        ObjectPropertyAxiom,
        IObjectPropertyDomain
    {
        private readonly IClassExpression _domain;

        public ObjectPropertyDomain(
            IOntology                 ontology,
            IObjectPropertyExpression objectPropertyExpression,
            IClassExpression          domain
            ) : base(
                ontology,
                objectPropertyExpression)
        {
            _domain = domain;
        }

        IClassExpression IObjectPropertyDomain.Domain => _domain;
    }
}

File: Web/ClientApp/src/app/Ontology/ObjectPropertyDomain.ts
import { IAxiomVisitor } from "./IAxiomVisitor";
import { IClassExpression } from "./IClassExpression";
import { IObjectPropertyDomain } from "./IObjectPropertyDomain";
import { IOntology } from "./IOntology";
import { IObjectPropertyExpression } from "./IPropertyExpression";
import { ObjectPropertyAxiom } from "./ObjectPropertyAxiom";

export class ObjectPropertyDomain
    extends ObjectPropertyAxiom
    implements IObjectPropertyDomain
{
    constructor(
        ontology                       : IOntology,
        public ObjectPropertyExpression: IObjectPropertyExpression,
        public Domain                  : IClassExpression
        )
    {
        super(ontology);
    }

    Accept(
        visitor: IAxiomVisitor
        )
    {
        visitor.ObjectPropertyDomain(this);
    }
}
